core/src/main/java/org/elasticsearch/transport/RequestHandlerRegistry.java

```diff
@@ -52,8 +55,21 @@ public Request newRequest() {
             return requestFactory.get();
     }
 
-    public TransportRequestHandler<Request> getHandler() {
-        return handler;
+    public void processMessageReceived(Request request, TransportChannel channel) throws Exception {
+        final Task task = taskManager.register(channel.getChannelType(), action, request);
+        if (task == null) {
+            handler.messageReceived(request, channel);
```

Comment:
Added a test when this is possible. This is basically a task management opt-out mechanism.